utils.js

```diff
@@ -236,10 +236,21 @@ function _formatAttachment(attachment1, attachment2) {
   // Instead of having a bunch of if statements guarding every access to image_data,
   // we set it to empty object and use the fact that it'll return undefined.
   attachment2 = attachment2 || {id:"", image_data: {}};
-  var fileName = attachment1.filename
-  attachment1 = attachment1.mercury ? attachment1.mercury : attachment1
-
-  switch (attachment1.attach_type) {
+  var fileName = attachment1.filename;
+  attachment1 = attachment1.mercury ? attachment1.mercury : attachment1;
+  var blob = attachment1.blob_attachment;
+  var type = blob ? blob.__typename :  attachment1.attach_type;
+  if (! type)
+    type = attachment1.attach_type;
```

Comment:
It seems like this and the `var type = blob ? blob.__typename :  attachment1.attach_type;` above are redundant. Am I missing something?

Reply:
Ah, I guess it handles the case that blob exists but blob.__typename is null/undefined. Could do it in one line with `blob && blob.__typename ? blob.__typename :  attachment1.attach_type`, but I guess I'll leave that call to you.